src/main/java/fr/openent/competences/service/impl/DefaultUtilsService.java

```diff
@@ -1490,4 +1525,64 @@ public void getYearsArchive(String idStructure, String type, Handler<Either<Stri
         Sql.getInstance().prepared(query.toString(), params, Competences.DELIVERY_OPTIONS,
                 validResultHandler(handler));
     }
+
+    @Override
+    public void getSubTopicCoeff(String idEtablissement, String idClasse, Promise<List<SubTopic>> promise) {
+        subTopicService.getSubtopicServices(idEtablissement,idClasse,event -> {
+            if(event.isRight()){
+                List<SubTopic> subTopics= new ArrayList<>();
+                for(Object subTopicobj : event.right().getValue()){
```

Comment:
for(JsonObject subTopic : ((List<JsonObject>) event.right().getValue()))) {

}

Reply:
okey dommage !